trade-finance-manager-api/src/v1/controllers/amendment.controller.js

```diff
@@ -88,14 +89,16 @@ const createAmendmentTFMObject = async (amendmentId, facilityId, auditDetails) =
     // gets latest amendment value and dates
     const latestValueResponse = await api.getLatestCompletedAmendmentValue(facilityId);
     const latestCoverEndDateResponse = await api.getLatestCompletedAmendmentDate(facilityId);
+    const facility = await api.findOneFacility(facilityId);
 
     let tfmToAdd = {};
     // populates array with latest value/exposure and date/tenor values
     tfmToAdd = await addLatestAmendmentValue(tfmToAdd, latestValueResponse, facilityId);
     tfmToAdd = await addLatestAmendmentCoverEndDate(tfmToAdd, latestCoverEndDateResponse, facilityId);
 
     let latestFacilityEndDateResponse;
-    if (isTfmFacilityEndDateFeatureFlagEnabled()) {
+    const isFacilityEndDateEnabled = isTfmFacilityEndDateFeatureFlagEnabled() && isGefFacility(facility.facilitySnapshot.type);
```

Comment:
Are we able to use `facility.facilitySnapshot.isGef` here instead of `isGefFacility(...`?

Not sure if we've gone full circle here, but searching in the codebase, all other instances of `isTfmFacilityEndDateFeatureFlagEnabled() &&` use `facility.facilitySnapshot.isGef` so I think we should be aligned

Reply:
No we can't as this code is making a call via `api.findOneFacility` which does not run through the mapper (because the mapper is only used on the `getFacility` endpoint in tfm-api (which is called by tfm-ui), which is actually really surprising).

We can't be aligned in this PR, as these are two old parts of the code one calling the mapper the other not. The mapper is rather surprising in it modifies fields on the facility snapshot and this makes it unreliable when acting on facility information, so I'm keen not to add it anywhere else as it could have incredibly surprising side effects.

The real solution here is a wider change to 1. split the mapper into 2 mappers, one for ui changes and one for api changes to the object, 2. add these mappers at the relevant places in the code

Ideally we wouldn't modify fields in tfm-api for the sake of api use -- we should not need to, we should have a few endpoints in dtfs that does this heavy lifting for us ie `getFacilityWithLatestAmendments` or something to that nature, that computes this in one place (this is going to be a piece of tech debt we could address in the upcoming tfm/portal work).